src/frontend/http.rs

```diff
@@ -0,0 +1,82 @@
+use crate::storage::api::{Key, Value};
+use crate::storage::LSM;
+use futures::executor::block_on;
+use hyper::{Body, Request, Response, Server, StatusCode};
+use routerify::prelude::*;
+use routerify::{RequestInfo, Router, RouterService};
+use std::sync::{Arc, RwLock};
+use std::{convert::Infallible, net::SocketAddr};
+
+async fn get_handler(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+    let key: &String = req.param("key").unwrap();
+    let key = Key::from(key.clone());
+
+    let lsm = req.data::<Arc<RwLock<LSM>>>().unwrap();
+    let lsm = lsm.read().unwrap();
```

Comment:
where is `lsm` instantiated?

Reply:
https://github.com/ysono/pancake/pull/15/files#diff-51f622a727225cad9caed61cc103982c3a45ec6227624abe42a441d9c79d1ab3R61 here, during `Router::build()`.

Also, I don't think this needs to be a RwLock. Just Arc.

Reply:
ah i see!

Reply:
Actually we need RwLock.